db/db.go

```diff
@@ -0,0 +1,130 @@
+package db
+
+import (
+	"context"
+
+	sq "github.com/Masterminds/squirrel"
+	"github.com/jackc/pgx/v4"
+	"github.com/jackc/pgx/v4/pgxpool"
+)
+
+// DB database
+type DB struct {
+	pool *pgxpool.Pool
+	psql sq.StatementBuilderType
+}
+
+// New creates a new DB connection
+func New(pool *pgxpool.Pool, psql sq.StatementBuilderType) *DB {
+	return &DB{
+		pool: pool,
+		psql: psql,
+	}
+}
+
+// Post ...
+type Post struct {
+	ID    int    `json:"id"`
+	Title string `json:"title"`
+	Body  string `json:"body"`
+}
+
+// GetPosts fetches all posts
+func (db *DB) GetPosts() ([]Post, error) {
+	posts := []Post{}
+
+	sql, _, err := db.psql.Select("*").From("posts").ToSql()
+	if err != nil {
+		return []Post{}, err
+	}
+
+	rows, err := db.pool.Query(context.Background(), sql)
+	if err != nil {
+		return []Post{}, err
+	}
+
+	defer rows.Close()
+
+	for rows.Next() {
+		var p Post
+		rows.Scan(&p.ID, &p.Title, &p.Body)
```

Comment:
Scan returns an error, don't forget to check it
```suggestion
		if err := rows.Scan(&p.ID, &p.Title, &p.Body); err != nil {
			return nil, err
		}
```